File: fsharp/cars-assemble/CarsAssemble.fs
module CarsAssemble

let successRate (speed: int): float =
    if speed < 5 then 
        1.0
    elif speed < 9 then
        0.9
    elif speed < 10 then
        0.8
    else
        0.77

let productionRatePerHour (speed: int): float =
    let cars_per_hour = 221.0
    cars_per_hour * float speed  * successRate speed

let workingItemsPerMinute (speed: int): int =
    productionRatePerHour speed / 60.0
    |> int

File: java/cars-assemble/src/main/java/CarsAssemble.java
public class CarsAssemble {

    public double productionRatePerHour(int speed) {
        return 221 * speed * successRate(speed);
    }

    public int workingItemsPerMinute(int speed) {
        return (int)productionRatePerHour(speed) / 60;
    }

    private double successRate(int speed) {
        if (speed < 5) {
            return 1.00;
        } else if (speed < 9) {
            return 0.90;
        } else if (speed == 9) {
            return 0.80;
        } else {
            return 0.77;
        }
    }
}
